src/ripple/app/tests/Flow_test.cpp

```diff
@@ -512,6 +512,83 @@ struct Flow_test : public beast::unit_test::suite
         }
     }
 
+    void testLineQuality ()
+    {
+        testcase ("Line Quality");
+
+        using namespace jtx;
+        auto const alice = Account ("alice");
+        auto const bob = Account ("bob");
+        auto const carol = Account ("carol");
+        auto const dan = Account ("dan");
+        auto const USDA = alice["USD"];
+        auto const USDB = bob["USD"];
+        auto const USDC = carol["USD"];
+        auto const USDD = dan["USD"];
+
+        //   Dan -> Bob -> Alice -> Carol; vary bobDanQIn and bobAliceQOut
+        for (auto bobDanQIn : {80, 100, 120})
+            for (auto bobAliceQOut : {80, 100, 120})
+                for (auto&& f : {feature ("nullFeature"), featureFlowV2})
+                {
+                    if (f != featureFlowV2 && bobDanQIn < 100 && bobAliceQOut < 100)
+                        continue;  // Bug in flow v1
+                    Env env (*this, features (f));
+                    env.fund (XRP (10000), alice, bob, carol, dan);
+                    env (trust (bob, USDD (100)), qualityInPercent (bobDanQIn));
+                    env (trust (bob, USDA (100)),
+                        qualityOutPercent (bobAliceQOut));
+                    env (trust (bob, USDA (100)));
```

Comment:
On closer examination of this test, I have a question.  We write `trust (bob, USDA (100))...` two consecutive times.  The first time sets the `QualityOut`, the second time does not.  I believe that, since the second call does not pass the `QualityOut` parameter, that the second call has no effect.  Is that your understanding as well?  Is there a reason we're including the second call?  Thanks.